Sources/CvPlayer.cpp

```diff
@@ -5983,29 +5922,25 @@ void CvPlayer::updateTradeRoutes()
 {
 	PROFILE_FUNC()
 
-	CLLNode<int>* pCityNode;
-	CvCity* pLoopCity;
-	CvCity* pListCity;
-	CLinkList<int> cityList;
-	int iTotalTradeModifier;
-	int iLoop;
-
-	for (pLoopCity = firstCity(&iLoop); pLoopCity != NULL; pLoopCity = nextCity(&iLoop))
+	for (CvPlayer::city_iterator cityItr = beginCities(); cityItr != endCities(); ++cityItr)
 	{
+		CvCity* pLoopCity = *cityItr;
 		pLoopCity->clearTradeRoutes();
 	}
 
+	CLinkList<int> cityList;
```

Comment:
All this function could be reduced to this (reminder for me for later, I'm not asking you to do it):
```cpp
bool SortCityByTradeModifier(const CvCity* lhs, const CvCity* rhs) const 
{
    return lhs->totalTradeModifer() > rhs->totalTradeModifier();
}
std::vector<CvCity*> tradeOrderCities(beginCities(), endCities());
std::sort(tradeOrderCities.begin(), tradeOrderCities.end(), SortCityByTradeModifier);
std::for_each(tradeOrderCities.begin(), tradeOrderCities.end(), boost::bind(&CvCity::updateTradeRoutes, _1));
```